### ISSUE TITLE
Route class property is not defined inside RouterMiddleware class

### ISSUE THREAD

**Issue by peter279k** [ar2labs/wiring#14]
As title, when I use the `phpstan` to do code analysis work with `level 1`, I found this issue in code analysis report.

```BASH
php vendor/bin/phpstan analyse src tests --level=1 -c phpstan.neon --no-progress

 ------ ----------------------------------------------------------------------- 
  Line   src/Http/Middleware/RouterMiddleware.php                               
 ------ ----------------------------------------------------------------------- 
  18     Property Wiring\Http\Middleware\RouterMiddleware::$router has unknown  
         class Wiring\Http\Middleware\Route as its type.                        
  38     Parameter $router of method                                            
         Wiring\Http\Middleware\RouterMiddleware::__construct() has invalid     
         typehint type Wiring\Http\Middleware\Router.                           
 ------ ----------------------------------------------------------------------- 
```

It looks like the `Route` class and `$router` variable are undefined class property and it should be fixed.

**Comment by arcostasi** [comment]
> As title, when I use the phpstan to do code analysis work with level 1, I found this issue in code analysis report.
> php vendor/bin/phpstan analyse src tests --level=1 -c phpstan.neon --no-progress
> 
>  ------ ----------------------------------------------------------------------- 
>   Line   src/Http/Middleware/RouterMiddleware.php                               
>  ------ ----------------------------------------------------------------------- 
>   18     Property Wiring\Http\Middleware\RouterMiddleware::$router has unknown  
>          class Wiring\Http\Middleware\Route as its type.                        
>   38     Parameter $router of method                                            
>          Wiring\Http\Middleware\RouterMiddleware::__construct() has invalid     
>          typehint type Wiring\Http\Middleware\Router.                           
>  ------ ----------------------------------------------------------------------- 
> It looks like the Route class and $router variable are undefined class property and it should be fixed.

Hello Peter, I was already aware of this case, as now not exist a PSR for router, I don't want acoupling the class Route in core, but I'll be analyzing another alternative. Thank you for analysis.

**Comment by peter279k** [comment]
Hi @arcostasi, thanks for your reply/concern.

I hope we can resolve issue without adding the `Route` class.

**Comment by peter279k** [comment]
@arcostasi, after researching this about `Router` class, it seems that we need to customize/develop our `Router` class.

And it's also compatible with `PSR-7` standard.

The references are as follows:

- https://www.php-fig.org/psr/psr-15/meta/
- https://github.com/delolmo/symfony-router/blob/master/src/SymfonyRouterMiddleware.php

As you can see the `symfony-router` repository, the `Router` class is Symfony Routing component and it's compatible with PSR-7.

Using the `PSR-15` Router middle ware to dispatch the `PSR-7` Router.

**Comment by arcostasi** [comment]
Router property fixes in release [v2.0.8](https://github.com/ar2labs/wiring/commit/e2085b578139c2f74baf0306384fb9209fd671c7)
